### PR TITLE
Introduction to Cross platform vs native mobile development

### PR DESCRIPTION
**Pull request by Aviral-03**
## Pull Request: Adding New Page under Tech Stack for Introduction to Mobile Development

**Description:**
This pull request adds a new section titled "Introduction to Cross platform vs Native mobile development " to the Tech Stack folder. The section aims to provide valuable resources for CSC301 students to kickstart their mobile development journey. It includes links to online learning platforms, and development tools. These resources offer a comprehensive starting point for students to understand the difference between cross platform mobile development and native mobile development.

**Changes:**
- Added a new folder called Mobile_Development
- Added new file called content
- Added File Images, PDF for roadmaps

### CODE REVIEW

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+4. [Native Mobile Development](#native-mobile-development)
+5. [Advantages and Challenges of Native Development](#advantages-and-challenges-of-native-development)
+6. [Comparing Performance and User Experience](#comparing-performance-and-user-experience)
+7. [Examples of Products](#examples-of-products)
+8. [Developer Resources](#developer-resources)
+9. [References](#references)
+
+## Cross-Platform Development Frameworks
+
+Cross-platform development for mobile involves creating applications that can run seamlessly across multiple operating systems, such as iOS and Android, utilizing a single codebase. Through cross-platform frameworks, developers gain the ability to streamline the development process by writing code once and deploying it across various platforms. CSC301 students can benefit greatly from understanding cross-platform development as it is easier to implement for multiple famous platforms such as iOS and Android. Students with existing knowledge in web can easily start-off with React Native providing them with a clear transition.
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:17]: Consider providing a brief statistic or anecdote to grab readers' attention and highlight the significance of the topic in the current mobile app development landscape.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+4. Greater reach by targeting multiple platforms simultaneously
+
+### Challenges
+
+1. Performance limitations compared to native development
+2. Platform-specific nuances and limitations may require workarounds
+3. Dependency on third-party frameworks and libraries, which may introduce compatibility issues
+4. Limited access to platform-specific features and APIs may restrict functionality
+
+## Popular Cross-Platform Development Tools and Frameworks
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:35]: Include a brief mention of hybrid app development approaches, such as using web views within native apps, to provide a more comprehensive overview of cross-platform options.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  - Code sharing across platforms with shared projects or .NET Standard libraries
+  - Used by companies like UPS, Honeywell, and Olo for mobile app development
+
+## Native Mobile Development
+
+Native mobile development for iOS and Android involves creating applications specifically tailored for each platform using their respective programming languages and frameworks. For iOS, developers typically use Swift while for Android, Kotlin or Java are commonly employed. Notable IDEs for native development include Xcode for iOS and Android Studio for Android, providing robust toolsets for designing, debugging, and deploying applications. Understanding native development will help CSC301 students with the ability to harness platform specific capabilities fully, delivering high-performance, feature-rich mobile applications tailored to partners' needs.
+
+### Advantages and Challenges of Native Development
+
+#### Advantages
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:58]: Provide more specific examples or case studies to illustrate the advantages and challenges mentioned, enhancing the depth of understanding for readers.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+- **Flutter**:
+  - Developed by Google, offers a fast and expressive way to build native apps for iOS and Android from a single codebase
+  - Features a layered architecture with customizable widgets for building UIs
+  - Hot reload feature for rapid iteration and debugging
+  - Used by companies like Alibaba, Google Ads, and Reflectly for mobile app development
+- **Xamarin**:
+  - Acquired by Microsoft, allows building native apps for iOS, Android, and Windows using C# and .NET
+  - Integration with Visual Studio and support for native APIs
+  - Code sharing across platforms with shared projects or .NET Standard libraries
+  - Used by companies like UPS, Honeywell, and Olo for mobile app development
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:50]: Consider adding a brief comparison table highlighting key features or differences between the mentioned frameworks for quick reference and comparison.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+## Native Mobile Development
+
+Native mobile development for iOS and Android involves creating applications specifically tailored for each platform using their respective programming languages and frameworks. For iOS, developers typically use Swift while for Android, Kotlin or Java are commonly employed. Notable IDEs for native development include Xcode for iOS and Android Studio for Android, providing robust toolsets for designing, debugging, and deploying applications. Understanding native development will help CSC301 students with the ability to harness platform specific capabilities fully, delivering high-performance, feature-rich mobile applications tailored to partners' needs.
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:54]: Include a brief discussion on the importance of platform-specific design guidelines and user experience considerations in native app development to emphasize the advantages of native approaches.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+#### Challenges
+
+1. **Platform Dependency**: Developing separate codebases for iOS and Android requires additional time and resources compared to cross-platform development, as developers need to maintain platform-specific codebases and implement platform-specific features separately. CSC301 students need people with both domain knowledge otherwise, they won't be able to fulfill partner's requirement.
+2. **Learning Curve**: Learning platform-specific languages and frameworks, such as Swift for iOS and Kotlin for Android, may require additional time and effort, especially for developers new to mobile development.
+3. **Release Cycle**: Releasing updates and new features simultaneously across multiple platforms may pose challenges due to differences in release processes and approval times between app stores. This can be seen as a challenge for CSC301 students who have limited time for releasing a product.
+
+## Comparing Performance and User Experience
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:70]: Expand on the user experience aspect by discussing how platform-specific UI elements and interactions contribute to a more native feel in native apps, enhancing user satisfaction.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+- **Cross-platform Development**:
+  - Performance may be impacted due to additional layers of abstraction and interpretation required by cross-platform frameworks.
+  - Execution speed and resource consumption may vary depending on the framework used and the complexity of the application.
+- **Native Development**:
+  - Native apps generally offer better performance as they directly access platform-specific APIs and optimizations.
+  - Optimized for the specific hardware and software of each platform, leading to faster execution and responsiveness.
+
+### Testing Environments
+
+- **Cross-platform Development**:
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:83]: Provide insights into the challenges and considerations specific to testing cross-platform apps, such as ensuring consistency across different devices and OS versions.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+### Long-term implications for scalability and codebase maintenance
+
+- **Cross-platform Development**:
+  - Codebase maintenance is simplified as changes are applied universally across platforms.
+  - Scalability may be limited by the capabilities and constraints of the chosen cross-platform framework. This is one of the biggest advantages as down the line partner's platform may stop working if the libraries for cross-platform language are no longer supported.
+- **Native Development**:
+  - Scalability can be optimized for each platform individually, allowing for greater flexibility and performance optimization.
+  - Maintenance efforts may increase with separate codebases for iOS and Android, but platform-specific optimizations can lead to better long-term scalability. This point needs to be ensured during handover of the project so that future CSC301 students can take over and build more features upon the existing codebase.
+
+## Examples of Products
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:99]: Expand the list to include a wider range of apps, showcasing the diversity of successful implementations across industries and domains.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+  - React Native: Facebook, Instagram, Airbnb
+  - Flutter: Alibaba, Google Ads, Reflectly
+- **Native Development**:
+  - iOS Apps: Instagram, Airbnb, Spotify
+  - Android Apps: Google Maps, Instagram, WhatsApp
+
+
+## Developer Resources
+
+Finally, for CSC301 students to get started with the development process, here are some of the resources they could refer to:
```
Comment (JanelGilani) [inline Topics/Tech_Stacks/Mobile_Development/content.md:111]: Provide brief descriptions or recommendations for each listed resource to help students understand their relevance and applicability.

**Review by JanelGilani — COMMENTED** [review]
Overall, great content! I have added some actionable feedback for you, otherwise the wiki page is informative and easy to follow as a CSC301 student.

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 1 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:17]: Noted.

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 2 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:35]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 3 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:58]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 4 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:50]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 5 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:54]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 9 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:111]: Noted.

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 8 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:99]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 7 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:83]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md (hunk 6 above)
Comment (Aviral-03) [reply Topics/Tech_Stacks/Mobile_Development/content.md:70]: Noted

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+3. Simplified maintenance and updates with a single codebase
+4. Greater reach by targeting multiple platforms simultaneously
+
+### Challenges
+
+1. Performance limitations compared to native development
+2. Platform-specific nuances and limitations may require workarounds
+3. Dependency on third-party frameworks and libraries, which may introduce compatibility issues
+4. Limited access to platform-specific features and APIs may restrict functionality
+
```
Comment (ananyamjain) [inline Topics/Tech_Stacks/Mobile_Development/content.md:34]: Would it be relevant to add some lines about how to choose between cross-platform and native development based on project requirements? Including factors such as the application's complexity, performance requirements, and development timeline can guide students in making informed decisions.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+## Popular Cross-Platform Development Tools and Frameworks
```
Comment (ananyamjain) [inline Topics/Tech_Stacks/Mobile_Development/content.md:35]: Consider adding another column to the table which mentions learning curve associated with each framework. For instance, students with a background in JavaScript may find React Native more approachable, whereas those interested in learning a new language might be drawn to Flutter and Dart.

File: Topics/Tech_Stacks/Mobile_Development/content.md [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+### Long-term implications for scalability and codebase maintenance
+
+- **Cross-platform Development**:
+  - Codebase maintenance is simplified as changes are applied universally across platforms.
+  - Scalability may be limited by the capabilities and constraints of the chosen cross-platform framework. This is one of the biggest advantages as down the line partner's platform may stop working if the libraries for cross-platform language are no longer supported.
+- **Native Development**:
+  - Scalability can be optimized for each platform individually, allowing for greater flexibility and performance optimization.
+  - Maintenance efforts may increase with separate codebases for iOS and Android, but platform-specific optimizations can lead to better long-term scalability. This point needs to be ensured during handover of the project so that future CSC301 students can take over and build more features upon the existing codebase.
+
+### Hybrid Mobile Application Approach
```
Comment (ananyamjain) [inline Topics/Tech_Stacks/Mobile_Development/content.md:93]: Provide a clearer definition of what constitutes a "hybrid" app.

**Review by ananyamjain — CHANGES_REQUESTED** [review]

**Comment by ananyamjain** [comment]
Here are some suggested changes. Overall it very well organized and considers all details and comparisons of the two selected topics.

**Review by ananyamjain — COMMENTED** [review]
Looks good!

**Review by ananyamjain — APPROVED** [review]
Looks good!

**Pull request merged**